src/DocumentDB.ChangeFeedProcessor/PartitionManagement/EqualPartitionsBalancingStrategy.cs

```diff
@@ -60,11 +60,23 @@ public IEnumerable<ILease> SelectLeasesToTake(IEnumerable<ILease> allLeases)
 
             if (expiredLeases.Count > 0)
             {
-                return expiredLeases.Take(partitionsNeededForMe);
+                var leasesToTake = expiredLeases.Take(partitionsNeededForMe).ToList();
```

Comment:
Is ToList() really needed? foreach should be able to work with just IEnumerable...